clippy_lints/src/regex.rs

```diff
@@ -143,8 +144,22 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
 
 fn str_span(base: Span, c: regex_syntax::ast::Span, offset: u16) -> Span {
     let offset = u32::from(offset);
-    let end = base.lo() + BytePos(c.end.offset as u32 + offset);
-    let start = base.lo() + BytePos(c.start.offset as u32 + offset);
+    let end = base.lo() + BytePos(
+        c.end
+            .offset
+            .try_into()
```

Comment:
You can directly `expect` here and use `+` for the addition.

Reply:
My thinking here was that the lint is warning about casting `c.end.offset` (which is a `usize`) to a `u32`, so if we are fixing the lint (instead of just silencing it with an `#[allow]`) then we are worried about the case when the `c.end.offset` might be large. In that case, it seemed to make sense to use `checked_add()` instead of `+` to prevent accidental wrapping. If you want though I can change it to just use `+`, or maybe it makes sense to just `#[allow]` the cast if we suspect the offset will always be small compared to 32 bits.

Reply:
hm yea, I'd probably rather allow it then. This is probably impossible to reach